### ISSUE TITLE
Hordes Steal Zombies

### ISSUE THREAD

**Issue by Headjack** [CleverRaven/Cataclysm-DDA#23939]
### Expected behavior: 
Hordes (wander spawns) travel the map, while regular spawned zeds mill around wherever they were.

### Actual behavior: 
Seems that static spawned zeds get vacuumed up into the hordes, and disappear the same way.

This eventually leads to no or very few zombies being around, and intentionally placed zeds like firefighters at a firehouse or swimmers at a pool disappearing if you visit there regularly. Could we add some form of tag to distinguish wander spawned zombies as the only ones that should be removed by the mechanic that makes them vanish as they "move along"?

**Comment by Night-Pryanik** [comment]
I find this behavior is very fitting for the zombies: they should form hordes by accumulating every zombie they met.

**Comment by DracoGriffin** [comment]
Agree with Night-Pryanik. Zombies should be forming hordes, even if wandering hordes option is not enabled. It's one of the few "realistic" effects of zombies naturally creating larger and larger groups.

I think it's more of a luck thing that you are traveling in the wake of a horde, rather than a curse.

**Comment by Headjack** [comment]
Possibly so, but it makes the game a bit easy when you can wait for zeds to not be there and take whatever you like freely. It also diminishes the impact of things like having zombie children at the schools and swimmers at the pool, disables the possibility of finding ones you want to find, like technicians at electrical hubs, and leaves unguarded valuable spots like the FEMA camp or military bases.  

Plus, it raises the question as to why everywhere I _GO TO_ has zombies milling around, but everywhere I _AM NEAR REGULARLY_ clears them out eventually, even if I dont kill them. I agree that they should from groups, that's why I run wander spawns, but there should also be stragglers both from a mechanical and lore perspective; as I've explained, finding various zeds in normal places is important for immersion and balance. 

Maybe instead there might be a tag that makes them not be picked up by hordes, to be applied to "set piece" zombies and 50% of all static spawns, that way there is a balance between lingerers and joiners. It makes sense that zeds would have slightly different behavior, since various personalities are already shown to impact that, as in scientists and grenadiers.

**Comment by John-Candlebury** [comment]
Yes I also think this is a problem, it can also result in locations meant to be challenging being empty of monsters, and thus safe to loot.

Hospitals are possibly the worst offender of this.

**Comment by RedPine91** [comment]
Perhaps there should be a "static horde" type.  A static horde would vacuum up zombies just like a horde, and refuse to give zombies to other hordes, while staying directly over a special location (hospital, military camp, etc.)

In that case, waiting around would make things WORSE, as the special location could steal zombies from nearby hordes.

Lore wise, this would be explained by the special zombies at a special location having faint memories/instincts that keep them near their current location.

**Comment by BrianLefler** [comment]
Would love to implement something better, this bothers me as well, but I don't know what people want.  Would a proposal like below work for people?

Each static-spawn zombie has a random level of interest in joining a horde (say 5 possible behaviors, equally weighted).
1) Will always join a horde.
2) Will join a horde of 4+ zombies.
3) Will join a horde if outside.
4) Will join a horde of 4+ zombies if outside.
5) Will never join a horde.

This would leave 60-80% of hospital & firefighter zombies in place if you zipped by, while still ensuring that if you ran through town and drew a giant (statically-spawned) zombie mob into the street, 80% of those static zombies would join a horde and follow you out.

**Comment by kevingranade** [comment]
If you're interested in making hordes better, look at https://github.com/CleverRaven/Cataclysm-DDA/projects/3, which is some plans around making them work properly.  
tl;dr the current implementation is a dead end, we need to track individual zombies instead of trying to fake it with moving spawn groups.

**Comment by BrianLefler** [comment]
I can't commit to a large rewrite of hordes, but the idea of a band-aid is appealing.  I don't think hordes need to be fully re-implemented to be better on than off--it just needs to not totally replace the existing system, so that's why the idea blended behavior (sometimes horde, sometimes linger) is appealing.

I have a prototype of the above suggestion implemented here: https://github.com/BrianLefler/Cataclysm-DDA/commit/231be33bdef4281e9b8c2f60e72734934fa1e4c3

That commit seems to do what's on the tin; tho requiring a size of 4 in a horde turns out to be really restrictive (after running around a few locations to suck up zombies into hordes, my hordes contain 52 outdoor-only zombies but only 11 size 4+ zombies).

I don't really have a sense yet for how it plays; like whether it hordes still feels threatening when they're smaller.
